Extensions/TokenAcquisition.cs

```diff
@@ -224,6 +224,27 @@ public void AddAccountToCacheFromJwt(OpenIdConnect.TokenValidatedContext tokenVa
                                            tokenValidatedContext.HttpContext);
         }
 
+        /// <summary>
+        /// Removes the account associated with context.HttpContext.User from the MSAL.NET cache
+        /// </summary>
+        /// <param name="context">RedirectContext passed-in to a <see cref="OnRedirectToIdentityProviderForSignOut"/> 
+        /// Openidconnect event</param>
+        /// <returns></returns>
+        public async Task RemoveAccount(RedirectContext context)
+        {
+            var user = context.HttpContext.User;
+            var app = CreateApplication(context.HttpContext, user, context.Properties, AzureADDefaults.CookieScheme);
+            var account = await app.GetAccountAsync(context.HttpContext.User.GetMsalAccountId());
+
+            // Workaround for the guest account
+            if (account == null)
```

Comment:
Note what needs to be done for guest accounts in single tenant apps (tenantId is a GUID or domain in appsettings.json and user logs-in with a guest account)